installer/installer.go

```diff
@@ -0,0 +1,821 @@
+			if !se.Timestamp.After(after) {
+				continue
+			}
+			stackEventExists := false
+			for _, e := range stackEvents {
+				if *e.EventID == *se.EventID {
+					stackEventExists = true
+					break
+				}
+			}
+			if stackEventExists {
+				continue
+			}
+			stackEvents = append(stackEvents, se)
+			if se.ResourceType != nil && se.ResourceStatus != nil {
+				if *se.ResourceType == "AWS::CloudFormation::Stack" {
+					if strings.HasSuffix(*se.ResourceStatus, actionCompleteSuffix) {
+						if strings.HasPrefix(*se.ResourceStatus, action) {
+							isComplete = true
+						} else {
+							isFailed = true
+						}
+					} else if strings.HasSuffix(*se.ResourceStatus, actionFailureSuffix) {
+						isFailed = true
+					}
+				}
+				var desc string
+				if se.ResourceStatusReason != nil {
+					desc = " (" + *se.ResourceStatusReason + ")"
+				}
+				name := *se.ResourceType
+				if se.LogicalResourceID != nil {
+					name = name + "(" + *se.LogicalResourceID + ")"
+				}
+				s.HandleEvent(fmt.Sprintf("%s\t%s%s", name, *se.ResourceStatus, desc))
+			}
+		}
+		if res.NextToken != nil {
+			nextToken = res.NextToken
+			fetchStackEvents()
+		}
+
+		return nil
+	}
+
+	for {
+		if err := fetchStackEvents(); err != nil {
+			return err
+		}
+		if isComplete {
+			break
+		}
+		if isFailed {
+			return fmt.Errorf("Failed to create stack %s", s.StackName)
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	return nil
+}
+
+func (s *Stack) fetchStack() error {
+	stackID := aws.String(s.StackID)
+
+	s.HandleEvent("Fetching stack")
+	res, err := s.cf.DescribeStacks(&cloudformation.DescribeStacksInput{
+		StackName: stackID,
+	})
+	if err != nil {
+		return err
+	}
+	if len(res.Stacks) == 0 {
+		return errors.New("Stack does not exist")
+	}
+	s.Stack = &res.Stacks[0]
+	return nil
+}
+
+func (s *Stack) fetchStackOutputs() error {
+	s.fetchStack()
+
+	s.InstanceIPs = make([]string, 0, s.NumInstances)
+	for _, o := range s.Stack.Outputs {
+		v := *o.OutputValue
+		if strings.HasPrefix(*o.OutputKey, "IPAddress") {
+			s.InstanceIPs = append(s.InstanceIPs, v)
+		}
+		if *o.OutputKey == "DNSZoneID" {
+			s.DNSZoneID = v
+		}
+	}
+	if len(s.InstanceIPs) != s.NumInstances {
+		return fmt.Errorf("expected stack outputs to include %d instance IPs but found %d", s.NumInstances, len(s.InstanceIPs))
+	}
+	if s.DNSZoneID == "" {
+		return fmt.Errorf("stack outputs do not include DNSZoneID")
+	}
+
+	return nil
+}
+
+func (s *Stack) configureDNS() error {
+	s.HandleEvent("Configuring DNS")
+	r53 := route53.New(s.Creds, s.Region, nil)
+	res, err := r53.GetHostedZone(&route53.GetHostedZoneRequest{ID: aws.String(s.DNSZoneID)})
+	if err != nil {
+		return err
+	}
+	if err := s.Domain.Configure(res.DelegationSet.NameServers); err != nil {
+		return err
+	}
+
+	s.HandleEvent("Waiting for DNS to propagate")
+	for {
+		status, err := s.Domain.Status()
+		if err != nil {
+			return err
+		}
+		if status == "applied" {
+			break
+		}
+		time.Sleep(time.Second)
+	}
```

Comment:
This step should be done after bootstrap, it isn't required to bootstrap, so we just waste time if we do it here.